File: 150_evaluate_reverse_polish_notation/evaluate_reverse_polish_notation.java
import java.util.Stack;

class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<Integer>();
        for (int i = 0; i < tokens.length; i++) {
            if (tokens[i].equals("+")) {
                stack.push(stack.pop() + stack.pop());
            } else if (tokens[i].equals("-")) {
                int n1 = stack.pop();
                stack.push(stack.pop() - n1);
            } else if (tokens[i].equals("*")) {
                stack.push(stack.pop() * stack.pop());
            } else if (tokens[i].equals("/")) {
                int n1 = stack.pop();
                stack.push(stack.pop() / n1);
            } else {
                stack.push(Integer.parseInt(tokens[i]));
            }
        }
        return stack.pop();
    }
}

File: 150_evaluate_reverse_polish_notation/evaluate_reverse_polish_notation.py
from typing import List


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i not in '+-*/':
                stack.append(int(i))
            else:
                if i == '+':
                    stack.append(stack.pop() + stack.pop())
                elif i == '-':
                    n1 = stack.pop()
                    stack.append(stack.pop() - n1)
                elif i == '*':
                    stack.append(stack.pop() * stack.pop())
                elif i == '/':
                    n1 = stack.pop()
                    stack.append(int(stack.pop() / n1))
        return stack.pop()
